src/components/AttributeSection.vue

```diff
@@ -1,15 +1,19 @@
-<!--
-  Displays entire section of attributes
- receives events from child component and check if change is possible in resources
- if changes are possible, emit event to top component to make changes, otherwise don't
- -->
-
-<script>
+<script setup>
+/**
+ * Displays entire section of attributes
+ * receives events from child component and check if change is possible in resources
+ * if changes are possible, emit event to top component to make changes, otherwise don't
+ */
 import { statSectionMixin } from "./mixins/statSectionMixin";
 import { attributesAndSkillsMixin } from "./mixins/attributesAndSkillsMixin";
 import Stat from "./Stat.vue";
 import RestrictionState from "./RestrictionState.vue";
 
+const emit = defineEmits(['statSectionHover'])
+
+</script>
+
+<script>
 export default {
   mixins: [statSectionMixin, attributesAndSkillsMixin],
   components: { stat: Stat, restrictionState: RestrictionState },
```

Comment:
napriklad definicia komponenty tu sa mi zda konfliktna s definiciou v importoch, ... akoze sa to teraz robi trochu inac. Na druhu stranu, ked sa toto da prec, tak vznika konflikt, ze stat zacina s malym s nizsie a potom to v IDe zcervena, tak to asi bude lepsie takto

Reply:
alebo ja to mozem este zkusit prerobit

Reply:
podla https://v3.vuejs.org/api/sfc-script-setup.html#using-components

Reply:
tak som to dal prec